src/olympia/addons/views.py

```diff
@@ -788,9 +788,8 @@ def filter_queryset(self, queryset):
             try:
                 app = AddonAppFilterParam(
                     self.request).get_object_from_reverse_dict()
-                type_ = None
-                if 'type' in self.request.GET:
-                    type_ = AddonTypeFilterParam(self.request).get_value()
+                type_ = AddonTypeFilterParam(
+                    self.request, optional=True).get_value()
```

Comment:
Will most params not be optional? Why this is definitely better, it would be even less boilerplate without the `optional=True`. But I guess it depends whether most params will or won't be optional.